### PR TITLE
Remove some wrong mpool use in agg

### PR DESCRIPTION
**Pull request by nnsgmsone**
## What type of PR is this?

- [ ] API-change
- [X] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

1. remove an illegal free in the distinct aggregate
2.  remove two illegal alloc in the group_concat

### CODE REVIEW

**Review by m-schen — APPROVED** [review]

**Review by fengttt — COMMENTED** [review]
first of all, why do we need unmarshalbinsry at all?   when is marshal and unmarshall called?

**Comment by nnsgmsone** [comment]
Since it needs to be passed to another node, I don't know what better way to do it than to serialize and then deserialize @fengttt

**Review by fengttt — CHANGES_REQUESTED** [review]
Need to talk

File: pkg/sql/colexec/agg/distagg.go [hunk 1]
```diff
@@ -429,11 +429,10 @@ func (a *UnaryDistAgg[T1, T2]) UnmarshalBinary(data []byte) error {
 	setDistAggValues[T1, T2](a, a.otyp)
 	a.srcs = decode.Srcs
 	a.maps = make([]*hashmap.StrHashMap, len(a.srcs))
-	m := mpool.MustNewZeroNoFixed()
+	m := mpool.MustNewZeroNoFixed() // this is a hack, will remove later
```
Comment (fengttt) [inline pkg/sql/colexec/agg/distagg.go:432]: This code is so confusing.    Why do you need copyData?   The you copy data into data in line 427.
This mpool need to be deleted.   In this case, why not put it in the operator?   Finally why should we use Marshal/Unmarshal interface ...   Is a operator or operator state?

File: pkg/sql/colexec/agg/distagg.go [hunk 2]
```diff
@@ -429,11 +429,10 @@ func (a *UnaryDistAgg[T1, T2]) UnmarshalBinary(data []byte) error {
...
 	for i, src := range a.srcs {
 		mp, err := hashmap.NewStrMap(true, 0, 0, m)
 		if err != nil {
-			m.Free(data)
 			for j := 0; j < i; j++ {
 				a.maps[j].Free()
 			}
```
Comment (fengttt) [inline pkg/sql/colexec/agg/distagg.go:438]: Following code, why is MySQLString useful?    And finally, a.priv, what is that?

File: pkg/sql/colexec/multi_col/group_concat/group_concat.go [hunk 3]
```diff
@@ -84,19 +84,15 @@ func (gc *GroupConcat) MarshalBinary() (data []byte, err error) {
 
 // encoding.BinaryUnmarshaler
 func (gc *GroupConcat) UnmarshalBinary(data []byte) error {
+	var err error
+
 	eg := &EncodeGroupConcat{}
 	types.Decode(data, eg)
 	m := mpool.MustNewZeroNoFixed()
```
Comment (fengttt) [inline pkg/sql/colexec/multi_col/group_concat/group_concat.go:91]: Again, this must be deleted.

File: pkg/sql/colexec/multi_col/group_concat/group_concat.go [hunk 4]
```diff
@@ -84,19 +84,15 @@ func (gc *GroupConcat) MarshalBinary() (data []byte, err error) {
...
-	da1, err := m.Alloc(len(eg.InsertsStrData))
-	if err != nil {
-		return err
-	}
+	da1 := make([]byte, len(eg.InsertsStrData))
 	copy(da1, eg.InsertsStrData)
```
Comment (fengttt) [inline pkg/sql/colexec/multi_col/group_concat/group_concat.go:93]: Why do you need copy?

**Review by fengttt — COMMENTED** [review]
So many things wrong ...

**Pull request closed without merging**